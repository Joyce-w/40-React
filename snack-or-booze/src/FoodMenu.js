import React from "react";
import { Link } from "react-router-dom";
import "./FoodMenu.css";
import {
  Card,
  CardBody,
  CardTitle,
  CardText,
  ListGroup,
  ListGroupItem
} from "reactstrap";

function FoodMenu({ items, category }) {
  return (
    <section className="col-md-4">
      <Card>
        <CardBody>
          <CardTitle className="font-weight-bold text-center">
            {category} Menu
          </CardTitle>
          <CardText>
            <span>Some quick example text to build on the card title and make up the
            bulk of the card's content.</span>
          </CardText>
          <ListGroup>
            {/* map through corresponding food menu to create links to each item */}
            {items.map(snack => (
              <Link to={`/${category}/${snack.id}`} key={snack.id}>
                <ListGroupItem>{snack.name}</ListGroupItem>
              </Link>
            ))}
          </ListGroup>
        </CardBody>
      </Card>
    </section>
  );
}

export default FoodMenu;
