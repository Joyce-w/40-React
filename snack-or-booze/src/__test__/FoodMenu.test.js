import React from 'react'
import { render, fireEvent } from '@testing-library/react'
//use jest to test for jest matchers
import '@testing-library/jest-dom/extend-expect'
import FoodMenu from '../FoodMenu'
import SnackOrBoozeApi from "../Api"
import { MemoryRouter } from 'react-router-dom'



const getAPI = async() => {
    let drinks = await SnackOrBoozeApi.getDrinks();
    return drinks;
}

//smoke test
it("should render w/o crashing", async () => {
    //get api data
    let drinks = await getAPI();
    render(<MemoryRouter><FoodMenu items={drinks} category="drink" /></MemoryRouter>)
})

//snapshot 
//use async because api call is required
it("matches snapshot", async () => {
    //get api data
    let drinks = await getAPI();
    const {asFragment} = render(<MemoryRouter><FoodMenu items={drinks} category="drink" /></MemoryRouter>)
    expect(asFragment()).toMatchSnapshot();
})

//specialized testing
it("should display drink menu", async () => {
    //get api data
    let drinks = await getAPI();
    const { getByText } = render(<MemoryRouter><FoodMenu items={drinks} category="drink" /></MemoryRouter>)
    expect(getByText('Martini')).toBeInTheDocument();
    expect(getByText('Menu', {exact: false})).toBeInTheDocument();
});
