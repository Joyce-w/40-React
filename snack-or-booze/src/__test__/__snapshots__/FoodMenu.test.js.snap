// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <section
    class="col-md-4"
  >
    <div
      class="card"
    >
      <div
        class="card-body"
      >
        <h5
          class="font-weight-bold text-center card-title"
        >
          drink Menu
        </h5>
        <p
          class="card-text"
        >
          <span>
            Some quick example text to build on the card title and make up the bulk of the card's content.
          </span>
        </p>
        <ul
          class="list-group"
        >
          <a
            href="/drink/martini"
          >
            <li
              class="list-group-item"
            >
              Martini
            </li>
          </a>
          <a
            href="/drink/negroni"
          >
            <li
              class="list-group-item"
            >
              Negroni
            </li>
          </a>
          <a
            href="/drink/gin-and-tonic"
          >
            <li
              class="list-group-item"
            >
              Gin and Tonic
            </li>
          </a>
          <a
            href="/drink/diet-coke"
          >
            <li
              class="list-group-item"
            >
              Diet Coke
            </li>
          </a>
          <a
            href="/drink/Boba-Milk-tea"
          >
            <li
              class="list-group-item"
            >
              Boba Milk tea
            </li>
          </a>
        </ul>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
